src/control/ToolHandler.cpp

```diff
@@ -446,10 +481,17 @@ void ToolHandler::loadSettings() {
     }
 }
 
-void ToolHandler::pointCurrentToolToButtonTool() { this->currentTool = this->buttonSelectedTool; }
+void ToolHandler::pointCurrentToolToButtonTool(Button button) {
+    Tool* tool = getButtonTool(button);
+    this->activeTool = tool;
+
+    this->listener->toolColorChanged(false);
```

Comment:
Sure the color does never change?

Reply:
I had to look this up.

This is actually slightly misleading when one reads it the actual signature of that function is:

```cpp
virtual void toolColorChanged(bool userSelection);
```

And `userSelection` basically denotes whether the there is currently something selected for which this change in color should be applied.

Reply:
If you have a good idea on how this part can be refactored to be more readable I'd be happy to refactor it :smile: .

Reply:
Ahh okay. That is a really bad way of mixing those two things together. Better would be to refactor both functionalities into their own functions and call those which are required. Mixing multiple actions within one function leads to eaxactly this issue of not really knowing what a function does.

Reply:
True. That is a good approach. I'll refactor that part.

Reply:
Done :heavy_check_mark: